src/core/kernel/KernelInteractions.cpp

```diff
@@ -0,0 +1,295 @@
+#include <QObject>
+#include <QWidget>
+#include <QDesktopServices>
+#include "common/QvHelpers.hpp"
+#include "KernelInteractions.hpp"
+#include "core/connection/ConnectionIO.hpp"
+
+namespace Qv2ray::core::kernel
+{
+    bool V2rayKernelInstance::ValidateKernel(const QString &vCorePath, const QString &vAssetsPath, QString *message)
```

Comment:
again: try replace with `std::optional<QString>` as return type

Reply:
I think we need the bool as the return type to check if the check is passed.
Or use &bool parameter and optional string as the return type?